From 6acfbbe3fc5fa545db9089da4448091ce598de4c Mon Sep 17 00:00:00 2001
From: Rex Dieter <rdieter@math.unl.edu>
Date: Fri, 15 Apr 2016 08:43:53 -0500
Subject: [PATCH 8/8] +changelog for prior commit

---
 ChangeLog | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index a43ae3a..e50ad3c 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,8 @@
 === xdg-utils 1.1.2 (unreleased) ===
 
+2016-04-15 Rex Dieter <rdieter@fedoraproject.org>
+   * xdg-mime: properly handle varied ktraderclient5 output (BR94946)
+
 2016-02-16 Rex Dieter <rdieter@fedoraproject.org>
    * xdg-screensaver: support cinnamon-screensaver (BR92966)
 
-- 
1.9.5

