Index: scripts/xdg-email
===================================================================
--- scripts/xdg-email
+++ scripts/xdg-email   2011-01-07 10:50:05.805814271 +0100
@@ -224,7 +224,13 @@
 exit_failure_operation_impossible()
 {
     if [ $# -gt 0 ]; then
-        echo "xdg-email: $@" >&2
+           if which zenity > /dev/null 2> /dev/null; then
+                   zenity --error --text="$@"
+           elif which kdialog > /dev/null 2> /dev/null; then
+                   kdialog --error "$@"
+           else
+                   echo "xdg-email: $@" >&2
+           fi
     fi
 
     exit 3
@@ -716,7 +722,7 @@
     open_xfce "${mailto}"
     ;;
 
-    generic|lxde)
+    lxde)
     open_generic "${mailto}"
     ;;
 

