diff -up xdg-utils-1.1.3/scripts/xdg-open.in.omv~ xdg-utils-1.1.3/scripts/xdg-open.in
--- xdg-utils-1.1.3/scripts/xdg-open.in.omv~	2019-05-11 02:00:38.221035505 +0200
+++ xdg-utils-1.1.3/scripts/xdg-open.in	2019-05-11 02:06:44.767627326 +0200
@@ -436,9 +436,9 @@ open_generic()
 
     # if BROWSER variable is not set, check some well known browsers instead
     if [ x"$BROWSER" = x"" ]; then
-        BROWSER=www-browser:links2:elinks:links:lynx:w3m
+        BROWSER=www-browser:lynx:links2:elinks:links:w3m
         if has_display; then
-            BROWSER=x-www-browser:firefox:iceweasel:seamonkey:mozilla:epiphany:konqueror:chromium:chromium-browser:google-chrome:$BROWSER
+            BROWSER=x-www-browser:chromium-browser-stable:falkon:otter-browser:arora:firefox:iceweasel:seamonkey:mozilla:epiphany:konqueror:chromium:chromium-browser:google-chrome:$BROWSER
         fi
     fi
 
