Index: scripts/xdg-email
===================================================================
--- scripts/xdg-email
+++ scripts/xdg-email   2011-01-07 10:53:33.713814271 +0100
@@ -242,7 +242,13 @@
 exit_failure_operation_failed()
 {
     if [ $# -gt 0 ]; then
-        echo "xdg-email: $@" >&2
+           if which zenity > /dev/null 2> /dev/null; then
+                   zenity --error --text="$@"
+           elif which kdialog > /dev/null 2> /dev/null; then
+                   kdialog --error "$@"
+           else
+                   echo "xdg-email: $@" >&2
+           fi
     fi
 
     exit 4
@@ -492,13 +498,13 @@
         gvfs-open "$1"
     else
         DEBUG 1 "Running gnome-open \"$1\""
-        gnome-open "$1"
+       msg=$(gnome-open "$1" 2>&1)
     fi
 
     if [ $? -eq 0 ]; then
         exit_success
     else
-        exit_failure_operation_failed
+        exit_failure_operation_failed "$msg"
     fi
 }
 


