diff -Naur xdg-utils-1.1.0/scripts/xdg-email.in xdg-utils-1.1.0.tpg/scripts/xdg-email.in
--- xdg-utils-1.1.0/scripts/xdg-email.in	2015-10-04 19:30:54.000000000 +0000
+++ xdg-utils-1.1.0.tpg/scripts/xdg-email.in	2015-10-05 01:48:18.231639349 +0000
@@ -459,7 +459,7 @@
     open_xfce "${mailto}"
     ;;
 
-    generic)
+    generic|lxde|lxqt)
     open_generic "${mailto}"
     ;;
 
diff -Naur xdg-utils-1.1.0/scripts/xdg-file-dialog.in xdg-utils-1.1.0.tpg/scripts/xdg-file-dialog.in
--- xdg-utils-1.1.0/scripts/xdg-file-dialog.in	2015-10-04 19:30:54.000000000 +0000
+++ xdg-utils-1.1.0.tpg/scripts/xdg-file-dialog.in	2015-10-05 01:49:25.620451469 +0000
@@ -308,7 +308,7 @@
 
 if [ "$action" = "openfilename" ]; then
     case "$DE" in
-        kde)
+        kde|lxqt)
         open_kde "$filename"
         ;;
 
@@ -322,7 +322,7 @@
     esac
 elif [ "$action" = "openfilenamelist" ]; then
     case "$DE" in
-        kde)
+        kde|lxqt)
         open_multi_kde "$filename"
         ;;
 
@@ -336,7 +336,7 @@
     esac
 elif [ "$action" = "savefilename" ]; then
     case "$DE" in
-        kde)
+        kde|lxqt)
         save_kde "$filename"
         ;;
 
@@ -350,7 +350,7 @@
     esac
 elif [ "$action" = "directory" ]; then
     case "$DE" in
-        kde)
+        kde|lxqt)
         directory_kde "$filename"
         ;;
 
diff -Naur xdg-utils-1.1.0/scripts/xdg-open.in xdg-utils-1.1.0.tpg/scripts/xdg-open.in
--- xdg-utils-1.1.0/scripts/xdg-open.in	2015-10-04 19:30:54.000000000 +0000
+++ xdg-utils-1.1.0.tpg/scripts/xdg-open.in	2015-10-05 01:50:53.851824353 +0000
@@ -412,6 +412,32 @@
     fi
 }
 
+open_lxqt()
+{
+    # pcmanfm-qt only knows how to handle file:// urls and filepaths, it seems.
+    if (echo "$1" | grep -q '^file://' ||
+        ! echo "$1" | egrep -q '^[[:alpha:]+\.\-]+:')
+    then
+        local file="$1"
+
+        # handle relative paths
+        if ! echo "$file" | egrep -q '^(file://)?/'; then
+            file="$(pwd)/$file"
+        fi
+
+        pcmanfm-qt "$file"
+
+    else
+        open_generic "$1"
+    fi
+
+    if [ $? -eq 0 ]; then
+        exit_success
+    else
+        exit_failure_operation_failed
+    fi
+}
+
 [ x"$1" != x"" ] || exit_failure_syntax
 
 url=
@@ -480,6 +506,10 @@
     open_lxde "$url"
     ;;
 
+    lxqt)
+    open_lxqt "$url"
+    ;;
+
     enlightenment)
     open_enlightenment "$url"
     ;;
diff -Naur xdg-utils-1.1.0/scripts/xdg-su.in xdg-utils-1.1.0.tpg/scripts/xdg-su.in
--- xdg-utils-1.1.0/scripts/xdg-su.in	2015-10-04 19:30:54.000000000 +0000
+++ xdg-utils-1.1.0.tpg/scripts/xdg-su.in	2015-10-05 01:51:24.484634262 +0000
@@ -151,7 +151,7 @@
 fi
 
 case "$DE" in
-    kde)
+    kde|lxqt)
     su_kde
     ;;
 
diff -Naur xdg-utils-1.1.0/scripts/xdg-terminal.in xdg-utils-1.1.0.tpg/scripts/xdg-terminal.in
--- xdg-utils-1.1.0/scripts/xdg-terminal.in	2015-10-04 19:30:54.000000000 +0000
+++ xdg-utils-1.1.0.tpg/scripts/xdg-terminal.in	2015-10-05 01:52:25.022226357 +0000
@@ -176,6 +176,19 @@
     fi
 }
 
+terminal_lxqt()
+{
+    if which qterminal &>/dev/null; then
+        if [ x"$1" = x"" ]; then
+            qterminal
+        else
+            qterminal -e "$1"
+        fi
+    else
+        terminal_generic "$1"
+    fi
+}
+
 #[ x"$1" != x"" ] || exit_failure_syntax
 
 command=
@@ -232,6 +245,10 @@
     terminal_lxde "$command"
     ;;
 
+    lxqt)
+    terminal_lxqt "$command"
+    ;;
+
     generic)
     terminal_generic "$command"
     ;;
diff -Naur xdg-utils-1.1.0/scripts/xdg-utils-common.in xdg-utils-1.1.0.tpg/scripts/xdg-utils-common.in
--- xdg-utils-1.1.0/scripts/xdg-utils-common.in	2015-10-04 19:30:54.000000000 +0000
+++ xdg-utils-1.1.0.tpg/scripts/xdg-utils-common.in	2015-10-05 01:53:32.012994987 +0000
@@ -273,6 +273,9 @@
          LXDE)
            DE=lxde;
            ;;
+         LXQt|X-LXQt)
+           DE=lxqt;
+           ;;
          MATE)
            DE=mate;
            ;;
@@ -306,6 +309,9 @@
          LXDE|Lubuntu)
            DE=lxde; 
            ;;
+         *LXQt)
+           DE=lxqt;
+           ;;
          MATE)
            DE=mate;
            ;;
